Add gofmt and govet to run via golangci-lint

nithyanatarajan opened the pull request:
Add gofmt and govet to run via golangci-lint. With this, we can get rid of `fmt` and `vet` and use a universal linter in both ci (present already) and local


Signed-off-by: Nithya Natarajan <nithyanatarajn@gmail.com>

arunvelsriram commented:
Now `lint` task will run everything. So, do we need the `fmt`, `vet` make tasks now?

nithyanatarajan commented:
@arunvelsriram we can remove them from make tasks. I wanted to get an opinion before removing them. Shall I also make those changes?

jskswamy commented:
@nithyanatarajan  if `lint` target can run the other tasks, we can remove the other make targets

nithyanatarajan commented:
1. Modified the `.golangci.yml` to fail if there are format errors.
2. `make fmt` formats the code if any formatting needed while `make lint` fails the build if any formatting needed (use to -s option)
3. Removed `make vet` as it is redundant

nithyanatarajan commented:
@arunvelsriram @jskswamy Kindly review the changes

arunvelsriram reviewed (approved):
Looks good to me
